### ISSUE TITLE
XForms are not loaded in IE8 standards mode

### ISSUE THREAD

**Issue by fdintino** [sidewinderlabs/backplanejs#3]
### Imported from backplanejs Google Code [issue 3](http://code.google.com/p/backplanejs/issues/detail?id=3).

<table>
    <tr>
        <th>Reporter</th>
        <td><a href="https://github.com/philbooth">philbooth</a></td>
    </tr>
    <tr>
        <th>Date</th>
        <td>5 Nov 2009  9:50:50 AM UTC</td>
    </tr>
</table>

In IE8, if documents are set to use standards mode, the XForms loader doesn't 
work. The same documents can be seen to load correctly in quirks mode.

---

**Priority**: Medium
**Type**: Defect
**Browser**: IE

**Comment by backplane-import** [comment]
### Update by [markbirbeck](https://github.com/markbirbeck) on 5 Nov 2009 11:09:31 AM UTC

**Module**: XForms

**Comment by backplane-import** [comment]
### Update by [philbooth](https://github.com/philbooth) on 5 Nov 2009  1:18:00 PM UTC

Owner set to [philbooth](https://github.com/philbooth)

Ticket status set to **Started**

**Comment by backplane-import** [comment]
### Comment by [philbooth](https://github.com/philbooth) on 9 Nov 2009  4:44:58 PM UTC

Another way to load these documents in IE8, less drastic than forcing quirks mode, is 
to put the document into IE7 compatibility mode. This can be done by adding the 
following element to the top of the head, after <title> and <meta> elements, but before 
any other elements:

```
<meta http-equiv="X-UA-Compatible" content="IE=EmulateIE7" />
```

**Comment by backplane-import** [comment]
### Comment by [philbooth](https://github.com/philbooth) on 12 Nov 2009  3:53:32 PM UTC

I've finally managed to fix this issue, see the attached patch.

Tested against the unit test suite and the regression test suite in Internet Explorer 
8, Firefox 3.5 and Safari 4.

---

**Updates**

Ticket status set to **FixPending**

---

**Attachments**

[issue-0003.patch](https://github.com/sidewinderlabs/backplanejs/raw/gcode-import/issues/attachments/3/issue-0003.patch) (5.6 KB)

**Comment by backplane-import** [comment]
### Comment by [philbooth](https://github.com/philbooth) on 16 Nov 2009  6:06:45 PM UTC

Patch is also available at:

http://groups.google.com/group/ubiquity-xforms-eng-code/msg/4583d8546d697879

**Comment by backplane-import** [comment]
### Comment by [philbooth](https://github.com/philbooth) on 17 Nov 2009  1:45:11 PM UTC

Patch was committed to UXF trunk at:

http://code.google.com/p/ubiquity-xforms/source/detail?r=3174

**Comment by backplane-import** [comment]
### Comment by [markbirbeck](https://github.com/markbirbeck) on 6 Dec 2009  3:27:32 PM UTC

The fix was incorporated at 23e39eaad69832cb9f7840c81b5fe47e9c79380c (originally [4591bfadc9](http://code.google.com/p/backplanejs/source/detail?r=4591bfadc9)), and then merged to the main trunk at 
r3c442ed65.

---

**Updates**

Ticket status set to **Fixed**
